src/shell/bash.rs

```diff
@@ -51,4 +52,8 @@ impl Shell for Bash {
             autoload_hook = autoload_hook
         ))
     }
+
+    fn to_string(&self) -> String {
+        String::from("bash")
+    }
```

Comment:
I mentioned that we should probably not stringify them as it hurts the soundness of the functions (allows any kind of string). But if we can't do that, this function should probably be as:

```suggestion
    fn as_str(&self) -> &'static str {
        "bash"
    }
```

As there is no need for the extra allocation, we know that statically and we _want_ it to be static. Then we can implement `ToString` that clones the `self.as_str()` if necessary, but I don't believe it is.